networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/DiscoveryConfig.java

```diff
@@ -118,7 +118,8 @@ public DiscoveryConfig build() {
 
     private void initMissingDefaults() {
       if (minRandomlySelectedPeers.isEmpty()) {
-        minRandomlySelectedPeers = OptionalInt.of(Math.max(1, minPeers * 2 / 10));
+        minRandomlySelectedPeers =
+            OptionalInt.of(Math.min(maxPeers, Math.max(1, minPeers * 2 / 10)));
```

Comment:
tempting to break that Math.min into a function and write unit tests around it just to show the behaviours that we're expecting, it'd make the expected behaviours clearer and make sure we don't break it in the future...

Reply:
Since that `min` handles just `maxPeers == 0` I've simplified the expression here: 202a75b 
Does it look better now?